ArchitectureSpec.md

```diff
@@ -0,0 +1,74 @@
+# Vision Input Team:
+Handles image, video, and camera input and generates arrays of RGB or depth data
+## Responsibilities
+- Finding Camera parameters
+- Field of View (FoV)
+- Handle multiple cameras
+- Synchronization
+## Class structure
+4 ImageGenerator classes
+- `DepthDataGenerator (single frame depth filename, still image filename)`
+- `DepthLiveGenerator (Realsense camera port int)`
+- `ImageGenerator (single image frame filename)`
+- `VideoFileGenerator(filename)`
+- `VideoLiveGenerator(camera port input int)`
+## API
+### get_horizontal_fov()
+Returns float degrees
+
+### get_vertical_fov()
+Returns float degrees
+
+### generate()
+Gets data for one frame
+
+Returns: `tuple(rgb ndarray shape (x, y, 3), depth ndarray shape(x,y))`
+For sources with no depth data, return `tuple(ndarray, None)`
```

Comment:
This section should mention that the numpy arrays `generate()` returns have the y-value first and then the x-value--(y, x, rgb)--since that's how opencv reads and uses arrays to represent images and depth data.

Reply:
Does this also apply to the depth array? Maybe for consistency we should return that in (row, column) order as well. It's a simple .transpose() call if it isn't already that way.

Reply:
The depth array is currently the same, (y, x).